### ISSUE TITLE
wix-style-react not working properly with CRA

### ISSUE THREAD

**Issue by bilalesi** [wix/wix-style-react#5429]
i created a CRA project with latest React and react-dom , and add the package  wix-style-react last version  like the documentation guide in this link https://github.com/wix/wix-style-react/blob/master/docs/usage-without-yoshi.md but nothing work 

i eject the project and add the webpack.config.js configuration 
i installed Stylable version 1 (documentation guidelines)
there are three parts of webpack.config.js i modified 
i attached the webpack configuration file 
![plugins](https://user-images.githubusercontent.com/2632473/80860824-2d4b3600-8c1f-11ea-9460-fc4172bdd020.png)
![rules](https://user-images.githubusercontent.com/2632473/80860831-34724400-8c1f-11ea-8e29-5de366a2ca39.png)
![css](https://user-images.githubusercontent.com/2632473/80860863-71d6d180-8c1f-11ea-9b39-fb8d75cd10ed.png)

the error i get is the following:
TypeError: _Pagination_st_css__WEBPACK_IMPORTED_MODULE_15___default(...) is not a function

in the console , i got :
<img width="939" alt="Capture" src="https://user-images.githubusercontent.com/2632473/80861573-40143980-8c24-11ea-9e3c-9d70d2d90ff0.PNG">

i think the problem is with stylable integration in webpack or other loaders, any solution please

**Comment by EgoziE** [comment]
Thanks for opening this issue, we'll investigate this as soon as we can.

**Comment by ivanji** [comment]
Hi guys - any pointers on what we could be doing to fix this? I've tried already so many things and still have no luck using wix-style-react on any of my projects. A few colleagues have also attempted it and no luck. Just a working webpack.config file that is compatible with CRA or NextJS would be tremendously helpful. Thanks.

**Comment by EgoziE** [comment]
@ivanji Please try this guide I wrote (and leave comments if needed)
https://github.com/wix/wix-style-react/pull/5545

I just want to say this wasn't easy and there are some specific rules you have to follow.
I hope that in the future wix-style-react will work perfectly with CRA.
For now, this is the best I could do :)

**Comment by ivanji** [comment]
Thanks @EgoziE for the effort. Your example works well using using `stylable-create-app` but once NextJS is added the same issue happens as when starting with `create-next-app`.

NextJS has its own webpack config and the way to pass additional rules or config is explained here: https://nextjs.org/docs/api-reference/next.config.js/custom-webpack-config - nothing too out of the ordinary. 

We just need to append rules or plugins to it. In this case, I append the stylable plugin with some config options, which I understand is the only build tool necessary to parse .st.css files (that is until a proper webpack loader is implemented which I saw as an experimental branch).

Here's my webpack config (or next.config.js as is known in a NextJS project)
  
```
const path = require('path')
const { StylableWebpackPlugin } = require('@stylable/webpack-plugin')
module.exports = {
  // https://nextjs.org/docs/basic-features/built-in-css-support#customizing-sass-options
  sassOptions: {
    includePaths: [
      path.join(__dirname, 'node_modules/wix-animations'),
      path.join(__dirname, 'node_modules/wix-style-react'),
      path.join(__dirname, 'src')
    ]
  },
  webpack: (config, { buildId, dev, isServer, defaultLoaders, webpack }) => {
    // Note: we provide webpack above so you should not `require` it
    // Perform customizations to webpack config
    // Important: return the modified config
    const options = {
       outputCSS: true, // *If this is not set it seems like the .st.css file is not parsed and webpack complains about unexpected token in .root*
    }
    config.plugins.push(new StylableWebpackPlugin(options))
    config.module.rules.push(
      {
        test: /\.(png|jpg|gif)$/,
        use: [
          {
            loader: 'url-loader',
            options: {
              limit: 8192
            }
          }
        ]
      },
      {
        test: /\.(png|jpg|jpeg|gif|svg)$/,
        loader: 'file-loader'
      }
     
    )
    return config
  }
}
```
if OutputCSS is not set then it seems like the .st.css file is not parsed and webpack complains about unexpected token in .root - if set to true, then I get the following error:
`UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'outputAst' of undefined`

It's worth noting the way NextJS handles css. As explained here: https://nextjs.org/docs/basic-features/built-in-css-support and that out of the box NextJS **server renders** the app.

NextJS supports CSS modules for component level styling, any global styles must go under a file called _app.js. But with Stylable files, once it sees a .st.css files it believes is a global file as it ends in .css files. So even, if I add: `{
        test: /\.css$/,
        exclude: /\st\.css$/,
        use: ['style-loader', 'css-loader']
      }` , I still get: 
```
UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'outputAst' of undefined
    at StylableOptimizer.optimize (/Users/ivanji/work/wix-style-react-next/node_modules/@stylable/optimizer/cjs/stylable-optimizer.js:23:32)
    at StylableGenerator.afterTransform (/Users/ivanji/work/wix-style-react-next/node_modules/@stylable/webpack-plugin/cjs/stylable-generator.js:54:37)
    at StylableGenerator.toCSS (/Users/ivanji/work/wix-style-react-next/node_modules/@stylable/webpack-plugin/cjs/stylable-generator.js:41:14)
    at /Users/ivanji/work/wix-style-react-next/node_modules/@stylable/webpack-plugin/cjs/stylable-module-helpers.js:130:33
    at Array.map (<anonymous>)
    at Object.renderStaticCSS (/Users/ivanji/work/wix-style-react-next/node_modules/@stylable/webpack-plugin/cjs/stylable-module-helpers.js:126:39)
    at StylableWebpackPlugin.createChunkCSSBundle (/Users/ivanji/work/wix-style-react-next/node_modules/@stylable/webpack-plugin/cjs/stylable-webpack-plugin.js:248:62)
    at /Users/ivanji/work/wix-style-react-next/node_modules/@stylable/webpack-plugin/cjs/stylable-webpack-plugin.js:182:30
    at Array.forEach (<anonymous>)
    at /Users/ivanji/work/wix-style-react-next/node_modules/@stylable/webpack-plugin/cjs/stylable-webpack-plugin.js:181:28
    at SyncHook.eval [as call] (eval at create (/Users/ivanji/work/wix-style-react-next/node_modules/tapable/lib/HookCodeFactory.js:19:10), <anonymous>:7:1)
    at SyncHook.lazyCompileHook (/Users/ivanji/work/wix-style-react-next/node_modules/tapable/lib/Hook.js:154:20)
    at /Users/ivanji/work/wix-style-react-next/node_modules/webpack/lib/Compilation.js:1399:37
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Users/ivanji/work/wix-style-react-next/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at AsyncSeriesHook.lazyCompileHook (/Users/ivanji/work/wix-style-react-next/node_modules/tapable/lib/Hook.js:154:20)
    at Compilation.seal (/Users/ivanji/work/wix-style-react-next/node_modules/webpack/lib/Compilation.js:1342:27)
(node:99153) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
(node:99153) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Again, any help or direction you guys could provide would be greatly appreciated.

**Comment by EgoziE** [comment]
Honestly, I'm not familiar with NextJS 😕 
I think this kind of configuration requires the attention of Stylable guys.

**Comment by ivanji** [comment]
Thanks, I raised the issue with the Stylable guys. 

I realised that as wix-style-react currently stands it can't be integrated, as NextJS doesn't allow importing stylesheets that aren't scoped to a module/component. https://github.com/vercel/next.js/blob/master/errors/css-modules-npm.md 

In wsr, components have import statements to the .st.css or scss stylesheets which means it will trigger the above error. Most other libraries use a ThemeProvider and/or the classnames package within their src files OR provide a built package for all components ready to use - or even a compiled version of the stylesheets. (BTW, why is your dist folder is the same as src?)

No matter how enamoured I am with the components you guys created - I think it's time to move on.
